FluidSynthConfig.cmake.in

```diff
@@ -144,5 +144,58 @@ if(NOT FLUIDSYNTH_IS_SHARED)
   list(REMOVE_ITEM CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
 endif()
 
-# finally, include the targets file
-include("${CMAKE_CURRENT_LIST_DIR}/FluidSynthTargets.cmake")
+set(FluidSynth_known_comps static shared)
+set(FluidSynth_comp_static NO)
+set(FluidSynth_comp_shared NO)
+foreach (FluidSynth_comp IN LISTS ${CMAKE_FIND_PACKAGE_NAME}_FIND_COMPONENTS)
+    if (FluidSynth_comp IN_LIST FluidSynth_known_comps)
+        set(FluidSynth_comp_${FluidSynth_comp} YES)
+    else ()
+        set(${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE
+            "FluidSynth does not recognize component `${FluidSynth_comp}`.")
+        set(${CMAKE_FIND_PACKAGE_NAME}_FOUND FALSE)
+        return()
+    endif ()
+endforeach ()
+
+if (FluidSynth_comp_static AND FluidSynth_comp_shared)
+    set(${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE
+        "FluidSynth `static` and `shared` components are mutually exclusive.")
+    set(${CMAKE_FIND_PACKAGE_NAME}_FOUND FALSE)
+    return()
+endif ()
+
+set(FluidSynth_static_targets "${CMAKE_CURRENT_LIST_DIR}/FluidSynth-static-targets.cmake")
+set(FluidSynth_shared_targets "${CMAKE_CURRENT_LIST_DIR}/FluidSynth-shared-targets.cmake")
```

Comment:
Both files define a `FluidSynth::FluidSynth` imported target, one is shared and the other is static.

Reply:
Yeah, that's how it's supposed to work.

Reply:
Right, as exclusive components.

It fails on the following (uncommon, but possible) construct:
```cmake
find_package(FluidSynth REQUIRED CONFIG COMPONENTS shared)
find_package(FluidSynth REQUIRED CONFIG COMPONENTS static)
```
(the second one is ignored)


But looks good to me

Reply:
Yes, that is how it works. Quoting [the mentioned post](https://www.alexreinking.com/blog/building-a-dual-shared-and-static-library-with-cmake.html):
>It is an error to request both static and shared components. If a single build needs both, it may separate its targets into two directories and call find_package with different components in each one. Since imported targets are not global by default, this works without any intervention on our part.